controllers/articles.controller.js

```diff
@@ -12,13 +12,30 @@ exports.getArticles = (req, res, next) => {
 
 
 exports.getArticleById = (req, res, next) => {
-    const {article_id} = req.params
-    selectArticleById(article_id)
+    selectArticleById(req.params)
     .then((article) => {
-        res.status(200).send({article})
+        if(article.length === 0){
+            return Promise.reject({status: 404, msg : 'Not Found'})
+        } 
+        res.status(200).send({article: article[0]})
     })
     .catch((err) => {
         next(err)
     })
 }
 
+exports.patchArticleById = (req, res, next) => {
+
+    Promise.all([updateArticle(req.params, req.body), selectArticleById(req.params)])
```

Comment:
This check of selectArticle shouldn't be necessary, if updateArticle returns nothing, we can assume it doesn't exist. However, if you did want to invoke selectArticleById (which would also throw the relevant error if the id didn't exist) you wouldn't then need to throw an error on line 31 - 33, as this would already be thrown by selectArticleById